### ISSUE TITLE
Allow passing a block to `assert_that` in the "block slot"

### ISSUE THREAD

**Issue by jcredding** [redding/assert#300]
When using `assert_that` and then the `raises` assertion it seems like it would be nice to be able to pass the proc/block in the "block slot" of the `assert_that`. Something like:

```ruby
# This example:
assert_that(-> { do_something }).raises(AnError)

# becomes:
assert_that { do_something }.raises(AnError)
# or
assert_that {
  do_something
}.raises(AnError)
```

A *bonus* might be that you could use blocks for other assertions as well though I don't think I'd use this personally:

```ruby
assert_that { 2 + 2 }.equals(4)
```

**Comment by kellyredding** [comment]
@jcredding I like `assert_that` getting the _actual value_ from the block slot if a block is given - for all assertions. I don't think I'd use it personally, like you say, but I don't see a good reason to limit the functionality. Thanks.

**Comment by kellyredding** [comment]
Closed b/c of #303
